src/Adapter/Grid/Data/Factory/CartGridDataFactory.php

```diff
@@ -0,0 +1,127 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Grid\Data\Factory;
+
+use Cart;
+use Context;
+use Currency;
+use Customer;
+use PrestaShop\PrestaShop\Core\Grid\Data\Factory\GridDataFactoryInterface;
+use PrestaShop\PrestaShop\Core\Grid\Data\GridData;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollection;
+use PrestaShop\PrestaShop\Core\Grid\Record\RecordCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Decorates Doctrine data factory for carts
+ */
+final class CartGridDataFactory implements GridDataFactoryInterface
+{
+    /**
+     * @var GridDataFactoryInterface
+     */
+    private $doctrineDataFactory;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param GridDataFactoryInterface $doctrineDataFactory
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(
+        GridDataFactoryInterface $doctrineDataFactory,
+        TranslatorInterface $translator
+    ) {
+        $this->doctrineDataFactory = $doctrineDataFactory;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $data = $this->doctrineDataFactory->getData($searchCriteria);
+
+        $records = $this->applyModifications($data->getRecords());
+
+        return new GridData($records, $data->getRecordsTotal(), $data->getQuery());
+    }
+
+    /**
+     * @param RecordCollectionInterface $records
+     *
+     * @return RecordCollection
+     */
+    private function applyModifications(RecordCollectionInterface $records)
+    {
+        $modifiedRecords = [];
+
+        foreach ($records as $record) {
+            switch ($record['status']) {
+                case 'abandoned_cart':
+                    $record['status'] = $this->translator->trans(
+                        'Abandoned cart',
+                        [],
+                        'Admin.Orderscustomers.Feature'
+                    );
+                    break;
+                case 'not_ordered':
+                    $record['status'] = $this->translator->trans('Not ordered', [], 'Admin.Orderscustomers.Feature');
```

Comment:
I cannot find this string within the page, can you help me with its context? In which case is it displayed?

Reply:
It's displayed in Shopping carts list in "Order ID" column when cart is not ordered. In legacy page this text was not translated.

Reply:
Alright, thanks @sarjon. Does it refer to abandoned carts? I think it should match an order status (Shop Parameters > Order Settings > Statuses). If there is no order status corresponding to this state of order, perhaps we can use another word like _Abandoned_, _Not placed_, _On hold_ or _Pending_?

Reply:
It has been discussed with the team and it appears that it is not relevant to create an order status in this case, but we agree on picking up another wording. I would choose _Not placed_ instead of _Non ordered_.

Reply:
Done.